mysql-test/t/percona_heap_var.test

```diff
@@ -165,3 +165,44 @@ ALTER TABLE t1 ADD PRIMARY KEY (a, c);
 --error ER_TABLE_CANT_HANDLE_BLOB
 ALTER TABLE t1 ADD INDEX idx (c, a);
 DROP TABLE t1;
+
+create table `t1` (
+  `id` int(11) not null auto_increment,
+  `json_column` text,
+  primary key (`id`)
+) engine=memory;
+show create table t1;
+
+# PS-3940: Engine can be changed to innodb and back
+alter table  t1 engine=innodb;
+alter table  t1 engine=memory;
+
+# PS-3943: Adding additional columns won't change the key_block size
+alter table t1 modify json_column blob;
+alter table t1 modify json_column text;
+show create table t1;
+
+alter table t1 modify json_column blob;
+alter table t1 modify json_column text;
+show create table t1;
+
+alter table t1 modify json_column blob;
```

Comment:
Why 3x ?

Reply:
Without my change, the original create statement didn't have a key_block_size. After the first alters, a correctly sized key_block_size (256) appeared, and it increased with the later alters - so it was different each time.

Reply:
Right, then please add comments "with the bug present, key_block_size now would be ... " or similar